x-pack/plugins/cloud_security_posture/public/components/fleet_extensions/aws_credentials_form.tsx

```diff
@@ -308,3 +429,60 @@ const AwsInputVarFields = ({
     ))}
   </div>
 );
+
+/**
+ * Update CloudFormation template and stack name in the Agent Policy
+ * based on the selected policy template
+ */
+const useCloudFormationTemplate = ({
+  packageInfo,
+  newPolicy,
+  updatePolicy,
+  setupFormat,
+}: {
+  packageInfo: PackageInfo;
+  newPolicy: NewPackagePolicy;
+  updatePolicy: (policy: NewPackagePolicy) => void;
+  setupFormat: SetupFormat;
+}) => {
+  useEffect(() => {
+    const checkCurrentTemplate = newPolicy?.inputs?.find((i: any) => i.type === CLOUDBEAT_AWS)
+      ?.config?.cloud_formation_template_url?.value;
+
+    if (setupFormat !== 'cloudFormation') {
+      if (checkCurrentTemplate !== null) {
+        updateCloudFormationPolicyTemplate(newPolicy, updatePolicy, null);
+      }
+      return;
+    }
```

Comment:
This can return `undefined`
```typescript
const checkCurrentTemplate = newPolicy?.inputs?.find((i: any) => i.type === CLOUDBEAT_AWS)
      ?.config?.cloud_formation_template_url?.value;
```

so this: 
```typescript
if (checkCurrentTemplate !== null) {
        updateCloudFormationPolicyTemplate(newPolicy, updatePolicy, null);
      }
```
essentially checks for `undefined !== null` which will result in true, and `updateCloudFormationPolicyTemplate` will get called.